Regular Expressions/RegExp (Basic).sublime-syntax

```diff
@@ -90,84 +87,113 @@ contexts:
-    # Deactivates 'x' mode
-    - match: (\?[ims]*-[ims]*x[imxs]*)(\))
-      captures:
-        1: storage.modifier.mode.regexp
-        2: keyword.control.group.regexp
-      set: [base-group, maybe-unexpected-quantifiers]
-    # Groups without 'x' mode
-    - match: '\?[ims]*-[ims]*x[imxs]*:'
+    # Groups which deactivate 'x' mode
+    - match: '{{deactivate_x_mode}}:'
       scope: storage.modifier.mode.regexp
+      pop: 3
       set: [group-body, maybe-unexpected-quantifiers]
     # Other modifiers
-    - match: '\?(?:[ixms]*-)?[ims]+:'
+    - match: '{{other_modifiers}}:'
       scope: storage.modifier.mode.regexp
-      set: [group-body-extended, maybe-unexpected-quantifiers]
-    - include: group-start-common
-    - match: ''
-      set: [group-body-extended, maybe-unexpected-quantifiers]
-
-  group-start-common:
+      pop: 1
     # Other modifiers (must come after other modifier matches)
-    - match: (\?(?:[ixms]*-)?[ixms]+)(\))
+    - match: ({{other_modifiers}})(\))
       captures:
         1: storage.modifier.mode.regexp
         2: keyword.control.group.regexp
+      pop: 3
+    - match: ''
       pop: 1
 
   group-body:
     - meta_content_scope: meta.group.regexp
     - match: \)
       scope: meta.group.regexp keyword.control.group.regexp
       pop: 1
-    - include: base-literal
+    - include: ingroup-xmode-on
+    - include: base
+    - include: literals
 
   group-body-extended:
     - meta_content_scope: meta.group.extended.regexp
     - match: \)
       scope: meta.group.extended.regexp keyword.control.group.regexp
       pop: 1
-    - include: base-literal-extended
+    - include: ingroup-xmode-off
+    - include: base-extended
+    - include: literals
+
+  ingroup-xmode-on:
+    # Activates 'x' mode
+    - match: (\()({{activate_x_mode}})(\))
+      scope: meta.group.regexp
+      captures:
+        1: keyword.control.group.regexp
+        2: storage.modifier.mode.regexp
+        3: keyword.control.group.regexp
+      set: [group-body-extended, maybe-unexpected-quantifiers]
+
+  ingroup-xmode-off:
+    # Deactivates 'x' mode
+    - match: (\()({{deactivate_x_mode}})(\))
+      scope: meta.group.extended.regexp
+      captures:
+        1: keyword.control.group.regexp
+        2: storage.modifier.mode.regexp
+        3: keyword.control.group.regexp
+      set: [group-body, maybe-unexpected-quantifiers]
+
+  xmode-on:
+    # Activates 'x' mode
+    - match: (\()({{activate_x_mode}})(\))
+      scope: meta.group.regexp
+      captures:
+        1: keyword.control.group.regexp
+        2: storage.modifier.mode.regexp
+        3: keyword.control.group.regexp
+      set: [base-literal-extended, maybe-unexpected-quantifiers]
+    # Deactivates 'x' mode, but already not active
+    - match: (\()({{deactivate_x_mode}})(\))
+      scope: meta.group.regexp
+      captures:
+        1: keyword.control.group.regexp
+        2: storage.modifier.mode.regexp
+        3: keyword.control.group.regexp
```

Comment:
Is that one still needed? If yes, would it need to be copied to `ingroup-xmode-on`?

It seems to be handled by `other_modifiers` in `group-start`.

Reply:
you're quite right, it's not needed here any more :)